### ISSUE TITLE
updateBranchProtection not working

### ISSUE THREAD

**Issue by mattdavis0351** [github/learning-lab-components#147]
## Bug Report

**Current behavior**
In the `before:` step updating branch protection works as expected

```yaml
before:
  - type: updateBranchProtection
```

Later in the course re-enabling it fails and prevents the step from running, which prevented the `createIssue` action from running and the course was unrecoverable.

```yaml
    event: pull_request.closed
    link: "{{ repoUrl }}/pull/2"
    actions:
      - type: updateBranchProtection
      - type: createIssue
```

**Reproduction**
Steps to reproduce the behavior in GitHub Learning Lab:
1. Enroll in `GitHub Actions: Using GitHub Script`
1. Follow course until you need to merge first PR
1. After merging PR you will not get a response

To test bug
1. Comment out the `updateBranchProtection` step
1. Enroll in `GitHub Actions: Using GitHub Script`
1. Follow course until you need to merge first PR
1. After merging PR you will a new issue is created

**Expected behavior**
A clear and concise description of what you expected to happen (with optional code snippets).

Branch protection for master would be re-enabled after user megerd

**Comment by hectorsector** [comment]
[Here's](https://lab.github.com/githubtraining/github-actions:-using-github-script/) the course being referenced.

**Comment by hectorsector** [comment]
I was ablate reproduce it in [my learner repo](https://github.com/hectorsector/write-github-script/pull/2), but no debug issue and no Sentry issues I can find. 

/cc @github/learning-engineering

**Comment by JasonEtco** [comment]
👋 thanks for the report y'all. I tested this out:

* `updateBranchProtection` in the `before` step correctly created the branch protection setting on `master`
* I was able to merge my pull request - _after_ refreshing the page to see that the "review required" message is gone.

So it seems to be working for me, but refreshing the page was a point of confusion. Would the bot approving the PR instead of removing/adding branch protections be an useful alternative? It'd avoid that issue with the PR not live-updating, and would be a more realistic flow for how PRs are typically gated.

**Comment by hectorsector** [comment]
Thanks for the response @JasonEtco. I think the issue isn't about the PR merging, but the next step occurring. I believe @mattdavis0351 has since made a few updates to the course to try and change to a PR review model, but [here's](https://github.com/githubtraining/write-github-script/blob/f2ab12f7ccbddb2ac0e72d7da0d98a37e9153c56/config.yml#L106-L115) the commit that I was trying it on.

**Comment by mattdavis0351** [comment]
@JasonEtco Correct, it's NOT  about the merging of the PR.  It was about being unable to call `updateBranchProtection` in the next step. 

I did remove these calls in favor of the `createReview` action as it seemed to provide us with the ability to use protected branches and still allow merges when we wanted them.
